src/middlewared/middlewared/plugins/disk_/wipe.py

```diff
@@ -25,16 +25,28 @@ def _wipe_impl(self, job, dev, mode, event):
                 # than that, ignore it.
                 return
 
-            # seek back to the beginning of the disk
-            os.lseek(f.fileno(), 0, os.SEEK_SET)
-
             # no reason to write more than 1MB at a time
             # or kernel will break them into smaller chunks
             if mode in ('QUICK', 'FULL'):
                 to_write = bytearray(CHUNK).zfill(0)
             else:
                 to_write = bytearray(os.urandom(CHUNK))
 
+            # The middle partitions often contain old cruft.  Clean those.
+            disk_parts = self.middleware.call_sync('disk.list_partitions', dev)
+            if len(disk_parts) > 1:
+                for partition in disk_parts[1:]:
+                    # Start 2 MiB back from the start and 'clean' 2 MiB past, 4 MiB total
+                    os.lseek(f.fileno(), partition['start'] - (2 * CHUNK), os.SEEK_SET)
+                    for i in range(4):
+                        os.write(f.fileno(), to_write)
+                        if event.is_set():
+                            return
+                        # This is quick. We can reasonably skip the progress update
```

Comment:
A couple of comments on this section
1. Does it only apply to 'QUICK' mode?  (i.e. non-quick mode is doing the whole disk, right?)
2. Should it be moved until after the HEAD and TAIL of the disk (though maybe read `disk_parts` first).  They _seem_ more important to scrub and I observe the `if event.is_set(): return`.  Hmm.